Assets/MRTK/Providers/HPMotionController/HPDeviceManager.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Collections.Generic;
+using Unity.Profiling;
+using UnityEngine;
+using UnityEngine.XR;
+using Microsoft.MixedReality.Input;
+
+namespace Microsoft.MixedReality.Toolkit.HP.Input
+{
+    public class MotionControllerState
+    {
+        public MotionControllerState(MotionController mc)
+        {
+            this.MotionController = mc;
+        }
+        public void Update(DateTime currentTime)
+        {
+            this.CurrentReading = MotionController.TryGetReadingAtTime(currentTime);
+        }
+        public MotionController MotionController { get; private set; }
+        public MotionControllerReading CurrentReading { get; private set; }
+
+    }
+
+    /// <summary>
+    /// Manages XR SDK devices.
+    /// </summary>
+    [MixedRealityDataProvider(
+        typeof(IMixedRealityInputSystem),
+        SupportedPlatforms.WindowsStandalone | SupportedPlatforms.MacStandalone | SupportedPlatforms.LinuxStandalone | SupportedPlatforms.WindowsUniversal,
+        "XRSDK Device Manager")]
+    public class HPDeviceManager : BaseInputDeviceManager, IMixedRealityCapabilityCheck
```

Comment:
Is there any way we can build this into the existing WMRProvider? I think both providers will raise events for both the existing WMR motion controller and the upcoming HP controller, so even this one isn't specific to the new hardware, just the new API.

Reply:
Or even make it an "either/or" situation where we keep the legacy provider and document that this new provider is needed for the new controller and this provider handles all WMR controllers. I feel like it'd be great if devs didn't need to change anything though!